### PR TITLE
r/dynamodb_tag: change equality check in propagation waiter

### PR DESCRIPTION
**Pull request by johnsonaj**
### Description

When multiple tag resources are added the wait function for tag propagation will continuously return `false` until timeout is reached.

```console
% make testacc TESTARGS='-run=TestAccDynamoDBTag_multipleTags' PKG=dynamodb

make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/dynamodb/... -v -count 1 -parallel 20  -run=TestAccDynamoDBTag_multipleTags -timeout 360m
=== RUN   TestAccDynamoDBTag_multipleTags
=== PAUSE TestAccDynamoDBTag_multipleTags
=== CONT  TestAccDynamoDBTag_multipleTags
    tag_test.go:142: Step 1/3 error: Error running apply: exit status 1

        Error: creating dynamodb resource (arn:aws:dynamodb:us-west-2:916215835482:table/tf-acc-test-4853890998756051563) tag (key1-2): waiting for resource (arn:aws:dynamodb:us-west-2:916215835482:table/tf-acc-test-4853890998756051563) tag propagation: timeout while waiting for state to become 'TRUE' (last state: 'FALSE', timeout: 2m0s)

          with aws_dynamodb_tag.test2,
          on terraform_plugin_test.tf line 34, in resource "aws_dynamodb_tag" "test2":
          34: resource "aws_dynamodb_tag" "test2" {


        Error: creating dynamodb resource (arn:aws:dynamodb:us-west-2:916215835482:table/tf-acc-test-4853890998756051563) tag (key1-3): waiting for resource (arn:aws:dynamodb:us-west-2:916215835482:table/tf-acc-test-4853890998756051563) tag propagation: timeout while waiting for state to become 'TRUE' (last state: 'FALSE', timeout: 2m0s)

          with aws_dynamodb_tag.test3,
          on terraform_plugin_test.tf line 40, in resource "aws_dynamodb_tag" "test3":
          40: resource "aws_dynamodb_tag" "test3" {

    panic.go:629: Error running post-test destroy, there may be dangling resources: exit status 1

        Error: deleting dynamodb resource (arn:aws:dynamodb:us-west-2:916215835482:table/tf-acc-test-4853890998756051563) tag (key1): waiting for resource (arn:aws:dynamodb:us-west-2:916215835482:table/tf-acc-test-4853890998756051563) tag propagation: timeout while waiting for state to become 'TRUE' (last state: 'FALSE', timeout: 2m0s)

--- FAIL: TestAccDynamoDBTag_multipleTags (265.35s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/dynamodb	272.118s
FAIL
make: *** [testacc] Error 1
```

### Relations

Relates #39326

### Output from Acceptance Testing

```console
% make testacc TESTARGS='-run=TestAccDynamoDBTable_tags\|TestAccDynamoDBTag_' PKG=dynamodb

make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/dynamodb/... -v -count 1 -parallel 20  -run=TestAccDynamoDBTable_tags\|estAccDynamoDBTag_ -timeout 360m
--- PASS: TestAccDynamoDBTag_disappears (52.15s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_emptyResourceTag (59.33s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_nullOverlappingResourceTag (59.44s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_nullNonOverlappingResourceTag (59.48s)
--- PASS: TestAccDynamoDBTag_basic (60.83s)
--- PASS: TestAccDynamoDBTable_tags_ComputedTag_OnCreate (61.09s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_emptyProviderOnlyTag (63.16s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_updateToResourceOnly (79.40s)
--- PASS: TestAccDynamoDBTag_value (87.93s)
--- PASS: TestAccDynamoDBTable_tags_AddOnUpdate (88.13s)
--- PASS: TestAccDynamoDBTable_tags_ComputedTag_OnUpdate_Add (88.73s)
--- PASS: TestAccDynamoDBTable_tags_ComputedTag_OnUpdate_Replace (88.77s)
--- PASS: TestAccDynamoDBTable_tags_EmptyTag_OnUpdate_Replace (90.10s)
--- PASS: TestAccDynamoDBTable_tags_EmptyTag_OnCreate (93.62s)
--- PASS: TestAccDynamoDBTable_tags_EmptyMap (49.03s)
--- PASS: TestAccDynamoDBTable_tags_null (44.87s)
--- PASS: TestAccDynamoDBTag_multipleTags (107.51s)
--- PASS: TestAccDynamoDBTable_tags_EmptyTag_OnUpdate_Add (107.82s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_updateToProviderOnly (51.96s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_nonOverlapping (114.46s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_overlapping (118.56s)
--- PASS: TestAccDynamoDBTable_tags (132.15s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_providerOnly (94.17s)
--- PASS: TestAccDynamoDBTag_ResourceARN_tableReplica (279.06s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/dynamodb	285.657s
```

### CODE REVIEW

File: internal/generate/tags/templates/v2/wait_tags_propagated_body.gtpl [hunk 1]
```diff
@@ -22,7 +22,7 @@ func {{ .WaitTagsPropagatedFunc }}(ctx context.Context, conn {{ .ClientType }},
 			output = output.IgnoreConfig(inContext.IgnoreConfig)
 		}
 
-		return output.Equal(tags), nil
+        return output.{{ .WaitFuncComparator }}(tags), nil
```
Comment (jar-b) [inline internal/generate/tags/templates/v2/wait_tags_propagated_body.gtpl:25]: Nice 👍

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc PKG=dynamodb TESTS=TestAccDynamoDBTag_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/dynamodb/... -v -count 1 -parallel 20 -run='TestAccDynamoDBTag_'  -timeout 360m

--- PASS: TestAccDynamoDBTag_disappears (29.42s)
--- PASS: TestAccDynamoDBTag_basic (31.94s)
--- PASS: TestAccDynamoDBTag_value (50.12s)
--- PASS: TestAccDynamoDBTag_multipleTags (71.76s)
--- PASS: TestAccDynamoDBTag_ResourceARN_tableReplica (197.23s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/dynamodb   204.015s
```

```console
% make testacc PKG=dynamodb TESTS="TestAccDynamoDBTable_Replica_tagsUpdate|TestAccDynamoDBTableReplica_basic|TestAccDynamoDBTableReplica_tableClass|TestAccDynamoDBTableReplica_tags"
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/dynamodb/... -v -count 1 -parallel 20 -run='TestAccDynamoDBTable_Replica_tagsUpdate|TestAccDynamoDBTableReplica_basic|TestAccDynamoDBTableRepl
ica_tableClass|TestAccDynamoDBTableReplica_tags'  -timeout 360m

--- PASS: TestAccDynamoDBTableReplica_basic (259.24s)
--- PASS: TestAccDynamoDBTableReplica_tags (311.55s)
--- PASS: TestAccDynamoDBTableReplica_tableClass (372.02s)
--- PASS: TestAccDynamoDBTable_Replica_tagsUpdate (431.19s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/dynamodb   436.395s
```

**Comment by jar-b** [comment]
Just needs a changelog.

File: internal/generate/tags/templates/v2/wait_tags_propagated_body.gtpl [hunk 2]
```diff
@@ -41,3 +23,29 @@ func {{ .WaitTagsPropagatedFunc }}(ctx context.Context, conn {{ .ClientType }},
 
 	return tfresource.WaitUntil(ctx, {{ .WaitTimeout }}, checkFunc, opts)
 }
+
+// checkFunc returns a function that checks if the tags are propagated.
+func checkFunc(ctx context.Context, conn {{ .ClientType }}, tags tftags.KeyValueTags, id string, optFns ...func(*{{ .AWSService }}.Options)) func() (bool, error) {
```
Comment (ewbankkit) [inline internal/generate/tags/templates/v2/wait_tags_propagated_body.gtpl:28]: `checkFunc` as a name seems a big vague and could well conflict with contributor-written code. How about something like `checkTagsPropagated`?

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc PKG=dynamodb TESTS=TestAccDynamoDBTag_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/dynamodb/... -v -count 1 -parallel 20 -run='TestAccDynamoDBTag_'  -timeout 360m

--- PASS: TestAccDynamoDBTag_disappears (29.61s)
--- PASS: TestAccDynamoDBTag_basic (32.12s)
--- PASS: TestAccDynamoDBTag_value (43.81s)
--- PASS: TestAccDynamoDBTag_multipleTags (78.89s)
--- PASS: TestAccDynamoDBTag_ResourceARN_tableReplica (185.82s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/dynamodb   191.120s
```


```console
% make testacc PKG=dynamodb TESTS=TestAccDynamoDBTable_tags
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.1 test ./internal/service/dynamodb/... -v -count 1 -parallel 20 -run='TestAccDynamoDBTable_tags'  -timeout 360m

--- PASS: TestAccDynamoDBTable_tags_DefaultTags_emptyResourceTag (67.35s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_nullNonOverlappingResourceTag (68.77s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_nullOverlappingResourceTag (68.82s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_emptyProviderOnlyTag (69.06s)
--- PASS: TestAccDynamoDBTable_tags_ComputedTag_OnCreate (69.22s)
--- PASS: TestAccDynamoDBTable_tags_null (74.51s)
--- PASS: TestAccDynamoDBTable_tags_EmptyMap (76.45s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_updateToResourceOnly (83.99s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_updateToProviderOnly (86.20s)
--- PASS: TestAccDynamoDBTable_tags_AddOnUpdate (87.75s)
--- PASS: TestAccDynamoDBTable_tags_ComputedTag_OnUpdate_Replace (91.05s)
--- PASS: TestAccDynamoDBTable_tags_EmptyTag_OnUpdate_Replace (93.37s)
--- PASS: TestAccDynamoDBTable_tags_ComputedTag_OnUpdate_Add (93.65s)
--- PASS: TestAccDynamoDBTable_tags_EmptyTag_OnCreate (97.06s)
--- PASS: TestAccDynamoDBTable_tags_EmptyTag_OnUpdate_Add (110.56s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_nonOverlapping (118.47s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_overlapping (120.41s)
--- PASS: TestAccDynamoDBTable_tags (131.51s)
--- PASS: TestAccDynamoDBTable_tags_DefaultTags_providerOnly (137.57s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/dynamodb   144.437s
```

**Pull request merged**